### PR TITLE
Added vectorized implementation for mat4.{scale, transpose, translate}

### PR DESCRIPTION
**Pull request by sajjadt**
Vectorized implementation for mat4.scale, mat4.transpose and mat4.translate

### CODE REVIEW

File: src/gl-matrix/mat4.js [hunk 1]
```diff
@@ -482,6 +566,49 @@ mat4.scale = function(out, a, v) {
...
+
+/**
+ * Scale the mat4 using SIMD if available and enabled
+ *
+ * @param {mat4} out the receiving matrix
+ * @param {mat4} a the matrix to scale
+ * @param {vec3} v the vec3 to scale the matrix by
+ * @returns {mat4} out
+ */
+mat4.scale = glMatrix.USE_SIMD ? mat4.SIMD.sclae : mat4.scalar.scale;
```
Comment (nickdesaulniers) [inline src/gl-matrix/mat4.js]: typo: `sclae`

**Comment by toji** [comment]
Looks good, but there's a merge conflict. Please rebase and I'll pull it in!

**Comment by nickdesaulniers** [comment]
tests!

**Comment by toji** [comment]
@nickdesaulniers: This should be automatically tested with the current mat4 test structure. 

Unfortunately I'm still seeing conflicts. Sorry! Don't know what the issue is.

**Comment by nickdesaulniers** [comment]
> This should be automatically tested with the current mat4 test structure. 

This seems to only be the case where `TEST_PATH` is only being used for multiply in the tests, not other functions like the ones being added in this PR.

I'll work with @sajjadt on resolving conflicts.

**Pull request closed without merging**